src/Jackalope/NodeType/NodeProcessor.php

```diff
@@ -73,45 +92,51 @@ public function __construct(
         $autoLastModified = true
     )
     {
-        $this->userId = $userId;
+        $this->userId = (string) $userId;
         $this->autoLastModified = $autoLastModified;
         $this->namespaces = $namespaces;
     }
 
     /**
-     * Process the given node.
+     * Process the given node and return eventual extra operations determined from the node.
      *
      * @param NodeInterface $node
+     *
+     * @return AddNodeOperation[] Additional operations that the client must execute for autocreated nodes.
```

Comment:
refactored to return the operations directly in process, instead of member variables.